dts/arm/nxp/nxp_rt5xx_common.dtsi

```diff
@@ -115,6 +116,7 @@
 	 */
 
 	clkctl0: clkctl@1000 {
+		/* FIXME This chip does NOT have a syscon */
```

Comment:
How is this handled in reality, then?

Reply:
right now the syscon clock control driver is technical debt its more of a shim to various platform specific SDK clock drivers which all have similar but different APIs. In the future, we should not use the same compatible for different types of IP. On this chip there is two "clkctl" IP, which is not the same as syscon, but probably has some heritage from the clock functionality of syscon IP, similar to how the rstctl probably has some heritage to the reset functions of syscon IP but is still much newer and different. (there is also a sysctl on this chip which is a 3rd split off of functionality not related to clocking or resetting)